scripts/build_wheel.py

```diff
@@ -165,6 +169,25 @@ def collect_setup_entries(
     return packages, package_data
 
 
+def verify_dependency(typeshed_dir: str, dependency: str, uploaded: str) -> None:
+    """Verify this is a valid dependency, i.e. a stub package uploaded by us."""
+    known_distributions = set(os.listdir(os.path.join(typeshed_dir, THIRD_PARTY_NAMESPACE)))
+    dependency = get_version.strip_dep_version(dependency)
+    assert dependency.startswith("types-"), "Currently only dependencies on stub packages are supported"
+    assert dependency[len("types-"):] in known_distributions, "Only dependencies on typeshed stubs are allowed"
+    with open(uploaded) as f:
+        uploaded_distributions = set(f.read().splitlines())
+    assert dependency in uploaded_distributions, f"{dependency} looks like a foreign distribution"
```

Comment:
Optional: Give a more specific error message if the dependency is different only because of upper/lower case differences?